pkg/resources/external_function_test.go

```diff
@@ -162,6 +131,103 @@ func TestExternalFunctionDelete(t *testing.T) {
 	})
 }
 
+func TestExternalFunctionUpdateComment(t *testing.T) {
+	// r := require.New(t)
+
+	// d := externalFunction(t, "database_name|schema_name|my_test_function|varchar", map[string]interface{}{"name": "my_test_function", "arg": []interface{}{map[string]interface{}{"name": "data", "type": "varchar"}}, "return_type": "variant", "comment": "mock comment"})
+
+	WithMockDb(t, func(db *sql.DB, mock sqlmock.Sqlmock) {
+		providers := providers()
+		providers["snowflake"].ConfigureFunc = func(s *schema.ResourceData) (interface{}, error) {
+			fmt.Println("HERE")
+			panic("foobar")
+			return db, nil
+		}
+
+		accName := "accName"
+
+		resource.Test(t, resource.TestCase{
+			Providers:  providers,
+			IsUnitTest: true, // we've mocked the DB
+			Steps: []resource.TestStep{
+				{
+					Config: externalFunctionConfig(accName, []string{"https://123456.execute-api.us-west-2.amazonaws.com/prod/"}, "https://123456.execute-api.us-west-2.amazonaws.com/prod/test_func"),
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr("snowflake_external_function.test_func", "name", accName),
+						resource.TestCheckResourceAttr("snowflake_external_function.test_func", "comment", "Terraform acceptance test"),
+						resource.TestCheckResourceAttrSet("snowflake_external_function.test_func", "created_on"),
+					),
+				},
+			},
+		})
+	})
+}
+
+// helpers
+// ------------------------------
+func expectExternalFunctionRead(mock sqlmock.Sqlmock, overrideShow map[string]string, overrideDescribe map[string]string) {
+	// order matters
+	columns := []string{"created_on", "name", "schema_name", "is_builtin", "is_aggregate", "is_ansi", "min_num_arguments", "max_num_arguments", "arguments", "description", "catalog_name", "is_table_function", "valid_for_clustering", "is_secure", "is_external_function", "language"}
+	defaultRowFields := []string{"now", "my_test_function", "schema_name", "N", "N", "N", "1", "1", "MY_TEST_FUNCTION(VARCHAR) RETURN VARCHAR", "mock comment", "database_name", "N", "N", "N", "Y", "EXTERNAL"}
+	defaultRow := zip(columns, defaultRowFields)
+
+	r := getRow(columns, defaultRow, overrideShow)
+
+	fmt.Printf("col %d, row %d", len(columns), len(r))
+	rows := sqlmock.NewRows(columns).AddRow(r...)
+	mock.ExpectQuery(`SHOW EXTERNAL FUNCTIONS LIKE 'my_test_function' IN SCHEMA "database_name"."schema_name"`).WillReturnRows(rows)
+
+	defaultDescribeRows := map[string]string{
+		"returns":         "VARIANT",
+		"null_handling":   "CALLED ON NULL INPUT",
+		"volatility":      "IMMUTABLE",
+		"body":            "https://123456.execute-api.us-west-2.amazonaws.com/prod/my_test_function",
+		"headers":         "{\"x-custom-header\":\"snowflake\"",
+		"context_headers": "[\"CURRENT_TIMESTAMP\"]",
+		"max_batch_rows":  "not set",
+		"compression":     "AUTO",
+	}
+
+	describeRows := sqlmock.NewRows([]string{"property", "value"})
+	for property, _ := range defaultDescribeRows {
```

Comment:
🚫 **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>S1005: unnecessary assignment to the blank identifier (gosimple)